src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -318,12 +318,46 @@ public IEnumerable<string> MakeCompliant(string name)
 
         private string CreateCompliantNameReusingPartialPrefixesAndSuffixes(string name)
         {
+            name = StripCommonPrefixes(name);
             name = EnsurePrefix(name);
             name = EnsureSuffix(name);
 
             return FinishFixingName(name);
         }
 
+        private static string StripCommonPrefixes(string name)
+        {
+            var index = 0;
+            while (name.Length > index + 1)
```

Comment:
nit: I kind of expected to see this expression inverted as I think it reads better. Totally not a big deal though. That's kind of in the eye of the beholder.

Reply:
Oh, I assumed from the form that `name` was getting written in the loop. Since it's not written, I definitely expected to see this the other way around.

Reply:
➡️ Fixed in 26fac2b0002c55f852b44438d1f9f79b84ee1b60